From 4e04afc2df9bbc26e5ab524b53a6f4f1e61d7c9e Mon Sep 17 00:00:00 2001
From: Javier Martinez Canillas <javierm@redhat.com>
Date: Tue, 5 Mar 2019 17:23:32 +0100
Subject: [PATCH] ucs2.h: fix logic that checks for UCS-2 string termination

Currently the loop to count the lenght of the UCS-2 string ends if either
of the two bytes are 0, but 0 is a valid value for UCS-2 character codes.

So only break the loop when 0 is the value for both UCS-2 char bytes.

Signed-off-by: Javier Martinez Canillas <javierm@redhat.com>
Signed-off-by: Peter Korsgaard <peter@korsgaard.com>
---
 src/ucs2.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/ucs2.h b/src/ucs2.h
index e0390c3..fd8b056 100644
--- a/src/ucs2.h
+++ b/src/ucs2.h
@@ -29,7 +29,7 @@ ucs2len(const void *vs, ssize_t limit)
 	const uint8_t *s8 = vs;
 
 	for (i = 0;
-	     i < (limit >= 0 ? limit : i+1) && s8[0] != 0 && s8[1] != 0;
+	     i < (limit >= 0 ? limit : i+1) && !(s8[0] == 0 && s8[1] == 0);
 	     i++, s8 += 2)
 		;
 	return i;
-- 
2.20.1

